glom/mutable.py

```diff
@@ -0,0 +1,52 @@
+'''
+this module contains Specs that perform mutations
+'''
+from .core import _TType, _T_PATHS, _t_child, Path, T, S, Spec, BaseSpec, glom
+
+
+# TODO: how to get this into default spec registry?
+# options:
+#   1- have a different registry here
+#   2- just eat the circularity
+#   3- move Assign back to core.py (but that is getting large)
+#   4- add some default behavior -- "if it is an instance of BaseSpec, call handle"
+class Assign(BaseSpec):
+    def __init__(self, path, val):
+        if isinstance(path, basestring):
+            path = Path(*path.split('.'))
+        if not isinstance(path, (Path, _TType)):
+            raise TypeError('path argument must be a . delimeted string, Path, T, or S')
+        if type(path) is Path:
+            path = path.path_t
+        segs = _T_PATHS[self.t]
+        if len(segs) < 3:
+            raise ValueError('path must have at least one element')
+        cur = segs[0]
+        assert cur in (T, S)
+        for i in range(1, len(segs) - 2, 2):
+            cur = _t_child(segs[i], segs[i + 1])
+        self.t = cur
+        self.op, self.arg = segs[-2:]
+        if self.op not in '[.P':
+            raise ValueError('last path of path must be setattr or setitem')
```

Comment:
I was surprised how much Assign ends up being very, very closely tied to Path / T / etc by the need to slice up the argument

one thing that comes to mind -- T having something like `dirname` / `filename` that lets the T be split to "base" and "leaf" segments would help a lot

also, perhaps assignment should be a simple function that associates with the T (the `Assign` spec should still be a thing; but it could leverage `set_t(t, val)` or something like this)